torch/csrc/jit/codegen/cuda/iter_visitor.cpp

```diff
@@ -97,11 +133,61 @@ void IterVisitor::traverseFrom(
   }
 }
 
+// This is a simplified version of traverseFrom
```

Comment:
simplified in what way? the name traverseFrom2() doesn't say much either.

Reply:
It should result in the same order. It is just temporary. I'm going to just keep the original one or the new one.

I find the original implementation has several places to improve mainly because part of the logic is replicated, e.g.,`handle` is called at multiple places. `next` is also called from two locations. That's what I wanted to clean up.

Reply:
great. I agree that the new version looks cleaner.